docs/core/04-contracts-overview.md

```diff
@@ -0,0 +1,117 @@
+| `DaoAuthorizableCloneable`       | Adds an internal initializer to `DaoAuthorizableBase`.                                                                                                           | used by `PluginCloneable`            |
+| `DaoAuthorizableBaseUpgradeable` | Provides a `DAO` variable, setter, and meta-transaction compatible modifier for upgradeable contracts to authorize function calls through an associated DAO.     | used by `DaoAuthorizableUpgradeable` |
+| `DaoAuthorizableUpgradeable`     | Adds an internal initializer method and storage gap to `DaoAuthorizableBaseUpgradeable`.                                                                         | used by `PluginUUPSUpgradeable`      |
+
+### Framework Infrastructure
+
+#### DAO Creation
+
+| Contract      | Description                                                | Relationship    |
+| :------------ | :--------------------------------------------------------- | :-------------- |
+| `DAOFactory`  | A template to setup plugins for a DAO using our framework. | creates `DAO`   |
+| `DAORegistry` | Registers DAOs                                             | registers `DAO` |
+
+#### Plugin Management
+
+| Contract             | Description                                                         | Relationship                                         |
+| :------------------- | :------------------------------------------------------------------ | :--------------------------------------------------- |
+| `PluginSetup`        | A template to setup plugins for a DAO using our framework.          | `Plugin`, `PluginCloneable`, `PluginUUPSUpgradeable` |
+| `PluginRepo`         | Versions a plugin by storing the associated `PluginSetup` contract. | versions `PluginSetup`                               |
+| `PluginRepoFactory`  | Creates `PluginRepo` contracts                                      | creates `PluginRepo`                                 |
+| `PluginRepoRegistry` | Registers `PluginRepo` contracts                                    | registers `PluginRepo`                               |
+
+### Plugin Implementations
+
+| Contract             | Description                                                                                  | Relationship                                                                    |
+| :------------------- | :------------------------------------------------------------------------------------------- | :------------------------------------------------------------------------------ |
+| `MerkleMinter`       | Mints `ERC-20` tokens and distributes them on merkle trees using `MerkleDistributor` clones. | is `PluginUUPSUpgradeable`, uses `MerkleDistributor`, is used by `TokenFactory` |
+| `MerkleDistributor`  | Distributes claimable ERC-20 tokens via a merkle tree.                                       | is `PluginUUPSUpgradeable`                                                      |
```

Comment:
Are they really a plugin? I see them as helpers of the ERC20 voting plugin (maybe below in utilities?)

Reply:
Yes, they are plugins and I would argue that they should stay as such when `TokenFactory` is refactored (https://aragonassociation.atlassian.net/browse/APP-1144).

Reply:
I have been looking at them as a helper of the ERC20 plugin since the beginning, but let's leave it aside and get moving :+1: